src/main/java/com/commercetools/sync/products/helpers/ProductReferenceResolver.java

```diff
@@ -205,20 +205,25 @@ public ProductReferenceResolver(@Nonnull final ProductSyncOptions productSyncOpt
 
         final Map<String, String> categoryOrderHintsMap = new HashMap<>();
         final CategoryOrderHints categoryOrderHints = draftBuilder.getCategoryOrderHints();
-
-        return categoryService.fetchMatchingCategoriesByKeys(categoryKeys)
-                              .thenApply(categories ->
-                                  categories.stream().map(category -> {
-                                      if (categoryOrderHints != null) {
-                                          ofNullable(categoryOrderHints.get(category.getKey()))
-                                              .ifPresent(orderHintValue -> categoryOrderHintsMap
-                                                  .put(category.getId(), orderHintValue));
-                                      }
-                                      return category.toReference();
-                                  }).collect(toList()))
-                              .thenApply(categoryReferences -> draftBuilder
-                                  .categories(categoryReferences)
-                                  .categoryOrderHints(CategoryOrderHints.of(categoryOrderHintsMap)));
+        try {
+            return categoryService.fetchMatchingCategoriesByKeys(categoryKeys)
+                    .thenApply(categories ->
+                            categories.stream().map(category -> {
+                                if (categoryOrderHints != null) {
+                                    ofNullable(categoryOrderHints.get(category.getKey()))
+                                            .ifPresent(orderHintValue -> categoryOrderHintsMap
+                                                    .put(category.getId(), orderHintValue));
+                                }
+                                return category.toReference();
+                            }).collect(toList()))
+                    .thenApply(categoryReferences -> draftBuilder
+                            .categories(categoryReferences)
+                            .categoryOrderHints(CategoryOrderHints.of(categoryOrderHintsMap)));
+        } catch (Exception exception) {
```

Comment:
How do you know the exception is related to reference? maybe NPE happened in that block of code ?

Reply:
Suppose to handle runtime exception, since categoryService.fetchMatchingCategoriesByKeys(...) goes to make query over network

Reply:
then why in catch block we're putting it into a Reference exception with a message ?

Reply:
Why do we need to catch here? See also my previous related comment.

Reply:
This change has been rolled back.